src/UBAFeeCalculator/UBAFeeSpokeCalculator.ts

```diff
@@ -167,61 +171,56 @@ export default class UBAFeeSpokeCalculator {
     return this.config.getBalanceTriggerThreshold(this.chainId, this.symbol);
   }
 
+  /**
+   * A convenience method for resolving the event fee for the spoke and the given symbol and a given flow type
+   * @param amount The amount of tokens to simulate
+   * @param flowType The flow type to simulate
+   * @param flowRange The range of the flow to simulate the event for. Defaults to undefined to simulate the event for the entire flow.
+   * @returns The event fee for the spoke and the given symbol and a given flow type
+   */
+  protected getEventFee(amount: BigNumber, flowType: "inflow" | "outflow", flowRange?: UBAFlowRange): FlowFee {
+    // The rough psuedocode for this function is as follows:
+    // We'll need two inflow/outflow curves
+    // We need to determine which flow curve to use based on the flow type
+    // Compute first balancing fee <- f(x, x+amnt)
+    // Compute second balance fee (oportunity cost) <- g(x+amnt, x)
+    // Incentive Fee (LP Fee Component): first balancing fee - second balance fee
+    // Return (LP Fee + Balancing Fee)
+    // #############################################
+
+    // We'll need to now compute the concept of the running balance of the spoke
+    const { runningBalance } = this.calculateHistoricalRunningBalance(flowRange?.startIndex, flowRange?.endIndex);
+
+    // We first need to resolve the inflow/outflow curves for the deposit and refund spoke
+    const flowCurve = this.config.getBalancingFeeTuples(this.chainId);
+
+    // Next, we'll need to compute the first balancing fee from the running balance of the spoke
+    // to the running balance of the spoke + the amount
+    const lpFee = computePiecewiseLinearFunction(flowCurve, runningBalance, amount.mul(flowType === "inflow" ? 1 : -1));
+
+    // We can now return the fee
+    return {
+      lpFee,
```

Comment:
nit: shouldn't this be the balancing fee portion of the LP fee? Then the LP fee would be computed in a separate module?

Reply:
Fair! Updated here: https://github.com/across-protocol/sdk-v2/pull/217/commits/7621170ce3786d57d86e238f862a9cbc56489f75